core/scripts/mtx-changer.in

```diff
@@ -131,166 +132,223 @@ wait_for_drive() {
-        ${MTX} -f $ctl inventory
-      fi
-      ${MTX} -f $ctl status >${TMPFILE}
-      rtn=$?
-      # can be converted to awk+sed+cut, see below
-      perl -ne '
-/Data Transfer Element (\d+):Empty/ && print "D:$1:E\n";
-/Data Transfer Element (\d+):Full \(Storage Element (\d+) Loaded\)(:VolumeTag =\s*(.+))?/ && print "D:$1:F:$2:$4\n";
-/Storage Element (\d+):Empty/ && print "S:$1:E\n";
-/Storage Element (\d+):Full( :VolumeTag=(.+))?/ && print "S:$1:F:$3\n";
-/Storage Element (\d+) IMPORT.EXPORT:Empty/ && print "I:$1:E\n";
-/Storage Element (\d+) IMPORT.EXPORT:Full( :VolumeTag=(.+))?/ && print "I:$1:F:$3\n";' ${TMPFILE}
-      # If perl isn't installed, you can use by those commands
-#cat ${TMPFILE} | grep "Data Transfer Element" | awk "{print \"D:\"\$4 \$7 \$9 \$10}" | sed "s/=/:/" | sed "s/Full/F:/" | sed "s/Empty/E/"
-#cat ${TMPFILE} | grep -v "Data Transfer Element" | grep "Storage Element" | grep -v "IMPORT/EXPORT" | awk "{print \"S:\"\$3 \$4 \$5}" | sed "s/IMPORT\/EXPORT//" | sed "s/Full *:VolumeTag=/F:/" | sed "s/Empty/E/"
-#cat ${TMPFILE} | grep -v "Data Transfer Element" | grep "Storage Element" | grep "IMPORT/EXPORT" | awk "{print \"I:\"\$3 \$4 \$5}" | sed "s/IMPORT\/EXPORT//" | sed "s/Full *:VolumeTag=/F:/" | sed "s/Empty/E/"
+  listall)
+    #  Drive content:	  D:Drive num:F:Slot loaded:Volume Name
+    #  D:0:F:2:vol2        or D:Drive num:E
+    #  D:1:F:42:vol42
+    #  D:3:E
+    #
+    #  Slot content:
+    #  S:1:F:vol1		  S:Slot num:F:Volume Name
+    #  S:2:E	       or S:Slot num:E
+    #  S:3:F:vol4
+    #
+    #  Import/Export tray slots:
+    #  I:10:F:vol10 	  I:Slot num:F:Volume Name
+    #  I:11:E	       or I:Slot num:E
+    #  I:12:F:vol40
 
-      rm -f ${TMPFILE} >/dev/null 2>&1
-      exit $rtn
-      ;;
+    debug "Doing mtx -f $ctl -- to list all"
+    make_temp_file
+    if test ${inventory} -ne 0; then
+      ${MTX} -f "${ctl}" inventory
+    fi
+    ${MTX} -f "${ctl}" status >"${TMPFILE}"
+    rtn=$?
+    # can be converted to awk+sed+cut, see below
+    #     perl -ne '
+    #/Data Transfer Element (\d+):Empty/ && print "D:$1:E\n";
+    #/Data Transfer Element (\d+):Full \(Storage Element (\d+) Loaded\)(:VolumeTag =\s*(.+))?/ && print "D:$1:F:$2:$4\n";
+    #/Storage Element (\d+):Empty/ && print "S:$1:E\n";
+    #/Storage Element (\d+):Full( :VolumeTag=(.+))?/ && print "S:$1:F:$3\n";
+    #/Storage Element (\d+) IMPORT.EXPORT:Empty/ && print "I:$1:E\n";
+    #/Storage Element (\d+) IMPORT.EXPORT:Full( :VolumeTag=(.+))?/ && print "I:$1:F:$3\n";' ${TMPFILE}
+    # If perl isn't installed, you can use by those commands
+    #cat ${TMPFILE} | grep "Data Transfer Element" | awk "{print \"D:\"\$4 \$7 \$9 \$10}" | sed "s/=/:/" | sed "s/Full/F:/" | sed "s/Empty/E/"
+    awk_code=$(
+cat << '_EOT_'
+    /^Data Transfer Element .*[0-9]:/ {
+      gsub("Data Transfer Element ","",$1)
+      gsub("Full .*",":F",$2)
+      gsub("Empty",":E",$2)
+      gsub("VolumeTag = ",":",$3)
+      print "D:" $1 $2 $3
+    }
```

Comment:
we still have a bug in here. The output for a drive with a volume loaded should look like this:
`D:<drive>:F:<slot-currently-loaded>:<volume-tag>`
e.g.
`D:0:F:3:000003L8` 
for volume 000003L8 loaded from slot 3 into drive 0

Reply:
Ok seen, I've fixed that now with gensub